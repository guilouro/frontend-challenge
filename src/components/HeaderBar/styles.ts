import styled, { css } from 'styled-components'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    height: 7rem;
    background-color: ${theme.colors.white};
    box-shadow: ${theme.shadow.medium};
    padding: ${theme.spacings.xxsmall};
    z-index: ${theme.layers.base};
    position: relative;
  `}
`
