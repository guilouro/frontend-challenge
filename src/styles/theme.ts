export default {
  grid: {
    container: '130rem',
    gutter: '3.2rem'
  },
  border: {
    radius: '1.6rem'
  },
  font: {
    family: 'Azo sans, sans-serif',
    light: 300,
    normal: 400,
    bold: 600,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '2.8rem',
      huge: '5.2rem'
    }
  },
  colors: {
    base: '#5f2eff',
    baseDark: '#5400db',
    mainBg: '#fafafa',
    white: '#FAFAFA',
    text: '#383743',
    textLight: '#6c6b76'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  },
  transition: {
    default: '0.5s ease-in-out',
    fast: '0.2s ease-in-out'
  }
} as const
