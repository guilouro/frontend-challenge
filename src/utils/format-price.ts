export default function formatPrice(price: number | bigint): string {
  return new Intl.NumberFormat('de-DE', {
    style: 'currency',
    currency: 'EUR'
  }).format(price)
}
