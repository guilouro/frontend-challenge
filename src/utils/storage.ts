import { CartObject } from 'hooks/use-cart'

const APP_KEY = 'BARKYN_CART'

export function getStorageItem() {
  if (typeof window === 'undefined') return

  const data = window.localStorage.getItem(APP_KEY)
  return JSON.parse(data || '[]')
}

export function setStorageItem(value: CartObject) {
  if (typeof window === 'undefined') return

  const data = JSON.stringify(value)
  return window.localStorage.setItem(APP_KEY, data)
}
