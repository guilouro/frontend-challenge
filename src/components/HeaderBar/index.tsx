import * as S from './styles'

export type HeaderBarProps = {
  children: React.ReactNode
}

const HeaderBar = ({ children }: HeaderBarProps) => (
  <S.Wrapper>{children}</S.Wrapper>
)

export default HeaderBar
