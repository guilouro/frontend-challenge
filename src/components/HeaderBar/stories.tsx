import { Story, Meta } from '@storybook/react'
import HeaderBar, { HeaderBarProps } from '.'

export default {
  title: 'HeaderBar',
  component: HeaderBar,
  args: {
    children: <h1>Barkyn</h1>
  },
  argTypes: {
    children: {
      type: ''
    }
  }
} as Meta

export const Default: Story<HeaderBarProps> = (args) => <HeaderBar {...args} />
