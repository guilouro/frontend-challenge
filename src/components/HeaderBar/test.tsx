import { render, screen } from 'utils/test-utils'
import theme from 'styles/theme'

import HeaderBar from '.'

describe('<HeaderBar />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <HeaderBar>
        <h1>Barkyn</h1>
      </HeaderBar>
    )

    expect(screen.getByRole('heading', { name: /barkyn/i })).toBeInTheDocument()
    expect(container.firstChild).toHaveStyleRule(
      'box-shadow',
      theme.shadow.medium
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
