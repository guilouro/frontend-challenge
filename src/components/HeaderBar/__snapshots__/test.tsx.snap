// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderBar /> should render the heading 1`] = `
.c0 {
  height: 7rem;
  background-color: #fff;
  box-shadow: 0px 2px 8px 0px rgba(18,18,23,0.08);
  padding: 0.8rem;
}

<header
  class="c0"
>
  <h1>
    Barkyn
  </h1>
</header>
`
